TSOClient/FSO.UI/Framework/UITween.cs

```diff
@@ -192,10 +455,18 @@ public void Stop()
 
         public void RenderPercent(float progress)
         {
-            foreach (var field in m_Fields)
+            for (int index = 0; index < m_Fields.Count; index++)
```

Comment:
I wanted the fields to be updated on the same thread, not on another thread since foreach creates a separate thread.

I thought it would fix issues where the completion and update actions were being triggered before the values were being updated, but this is still an issue.

That is to say it's not possible to check the progress of any field during an update action invocation; so I can't check a tween's progress on an object's y value, for example, by checking its y on that invocation.

Reply:
`foreach` does not create another thread, it is basically syntactic sugar for enumerating `IEnumerable<T>` types. It still iterates each item in order of first to last.

All UI actions are performed on the same thread. I suppose the issue you encountered was introduced by the queue system, so I don't know exactly what you mean by this. Just make sure invoking the events happens after the values are updated.